packages/app/src/state/savedGraphs.ts

```diff
@@ -135,7 +135,7 @@ export const projectPluginsState = selector({
 
 export type OpenedProjectInfo = {
   project: Project;
-  fsPath: string;
+  fsPath?: string | null;
```

Comment:
Does this need to be `string | null | undefined`? Can we choose either `null` or `undefined`?

Reply:
The code does use `null` in some spots.

```
      setOpenedProjects((openedProjects) => ({
        ...openedProjects,
        [currentProject.metadata.id]: {
          project: currentProject,
          fsPath: null,
        } satisfies OpenedProjectInfo,
      }));
```

On the subject of `nulls` in the code. There are several places where things like `x == null` or `x != null` are being used. These expressions are leveraging type coercion, and can be pretty confusing for junior developers. For example; `x == null` will be `true` if `x` is `undefined`. That can introduce bugs if the dev isn't aware of what's going on.

I would like to later, clean these up and make the expressions more explicit and not rely on coercion. In the meantime, I think it's best we document with the types that `null` and `undefined` are being used interchangeably in the code.

Reply:
@codemile It's an idiom we use at ironclad, `== null` and `!= null` are explicitly allowed, even though `null` isn't really. There's even an [estlint rule config](https://eslint.org/docs/latest/rules/eqeqeq#allow-null) for it. The idea it to treat `null` and `undefined` as basically the same thing, but so that you don't have to do `val === null || val === undefined` all over the places. This follows the same logic as `??`, which checks for null or undefined.